imports/client/components/ChatPeople.tsx

```diff
@@ -363,47 +250,29 @@ const ChatPeople = ({
         mediaSource = await navigator.mediaDevices.getUserMedia(mediaStreamConstraints);
       } catch (e) {
         setError(`Couldn't get local microphone: ${(e as Error).message}`);
-        setCallState(CallState.STREAM_ERROR);
+        callDispatch({ type: 'capture-error', error: e as Error });
         return;
       }
 
       const AudioContext = window.AudioContext ||
         (window as {webkitAudioContext?: AudioContext}).webkitAudioContext;
       const audioContext = new AudioContext();
 
-      setAudioState({
-        audioContext,
-        mediaSource,
+      callDispatch({
+        type: 'join-call',
+        audioState: {
+          mediaSource,
+          audioContext,
+        },
       });
-      setCallState(CallState.IN_CALL);
     } else {
-      setError('Couldn\'t get local microphone: browser denies access on non-HTTPS origins');
-      setCallState(CallState.STREAM_ERROR);
+      const msg = 'Couldn\'t get local microphone: browser denies access on non-HTTPS origins';
+      setError(msg);
+      callDispatch({ type: 'capture-error', error: new Error(msg) });
     }
-  }, []);
-
-  const leaveCall = useCallback(() => {
-    trace('ChatPeople leaveCall');
-    stopTracks(audioState.mediaSource);
-    setCallState(CallState.CHAT_ONLY);
-    setLocalAudioControls({
-      muted: false,
-      deafened: false,
-    });
-    setAudioState({
-      audioContext: undefined,
-      mediaSource: undefined,
-    });
-  }, [audioState]);
+  }, [callDispatch]);
 
-  useEffect(() => {
-    // When unmounting, stop any tracks that might be running
-    return () => {
-      trace('ChatPeople stop tracks on unmount');
-      // Stop any tracks that might be running.
-      stopTracks(audioState.mediaSource);
-    };
-  }, [audioState.mediaSource]);
+  // TODO: ensure we do proper cleanup of the sending track on unmount
```

Comment:
This is taken care of

Reply:
Specifically, by https://github.com/deathandmayhem/jolly-roger/pull/1023/commits/fb9d6c23d7f6a3a1c2870f7cad8703ad2f7a267a#diff-abb257e3f8aad9487024c922792f027ffc5ec9de52e60cadb72c6e439d6ef4a3R324-R333